### PR TITLE
feat: retain cli args when relaunching after update, closes #7402

### LINKED ISSUE #7402: [bug] relaunch application after installUpdate - the args are lose
### Describe the bug

relaunch application after installUpdate - the args are lose

### Reproduction

Declare cli args in tauri.conf.json
Open the application with args (e.g via Windows: cmd.exe /k AppName arg1 arg2)
Do relaunch - the args appear.
Do installUpdate and then relaunch - the args lose.

### Expected behavior

The args to stay after install updates

### Platform and versions

```text
[✔] Environment
    - OS: Windows 10.0.19045 X64
    ✔ WebView2: 114.0.1823.67
    ✔ MSVC: Visual Studio Professional 2022
    ✔ rustc: 1.67.0 (fc594f156 2023-01-24)
    ✔ Cargo: 1.67.0 (8ecd4f20a 2023-01-10)
    ✔ rustup: 1.25.2 (17db695f1 2023-02-01)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 18.14.0
    - npm: 9.3.1

[-] Packages
    - tauri [RUST]: 1.3.0
    - tauri-build [RUST]: 1.3.0   
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.2
    - @tauri-apps/api [NPM]: 1.3.0
    - @tauri-apps/cli [NPM]: 1.3.1

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist/apps/locota
    - devPath: http://localhost:4200/
    - framework: Angular
    - bundler: Webpack
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: .changes/tauri-bundler-nsis-args.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'tauri-bundler': 'minor:feat'
+---
+
+On Windows, NSIS isntaller now supports `/ARGS` flag to pass arguments to be used when launching the app after installation, only works if `/R` is used.
```
Comment (lucasfernog) [inline .changes/tauri-bundler-nsis-args.md]:
```suggestion
On Windows, NSIS installer now supports `/ARGS` flag to pass arguments to be used when launching the app after installation, only works if `/R` is used.
```

File: .changes/tauri-updater-retain-args.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'tauri': 'minor:enhance'
+---
+
+On Windows, retain command line args when relaunching the app after an update. Supports NSIS and WiX (wihtout elevated update task).
```
Comment (lucasfernog) [inline .changes/tauri-updater-retain-args.md]:
```suggestion
On Windows, retain command line args when relaunching the app after an update. Supports NSIS and WiX (without elevated update task).
```

**Comment by lucasfernog** [comment]
The argument list had a trailing `, ` (using join() is better) that was being denied by one of the bundle targets (probably wix?). Remember to test it with the updater e2e test: `cargo test --test '*' -- --ignored`.

**Comment by amrbashir** [comment]
the test kept failing on my end because Windows defender kept flagging it as a virus and removing it, so I pushed to see if it works on CI and forgot about it but I did test it manually with another app, however I didn't test after I switched the joining logic.

**Comment by amrbashir** [comment]
And I had to refrain from using `.join` because it is not implemented for `OsString` in 1.61

**Comment by lucasfernog** [comment]
This needs to be audited. @tweidinger @chippers can this be exploited? I'll go ahead and do the one change I know you'll request, which is pulling from the cached args instead of reading it at update time.

**Comment by lucasfernog** [comment]
@amrbashir can you add "change Env.args to pull from std::env::args_os instead of std::env::args" to the v2 breaking change list? :|

**Comment by lucasfernog** [comment]
```
Start-Process : Missing an argument for parameter 'ArgumentList'. Specify a parameter of type 'System.String[]' and try again.
At line:1 char:276
+ ... projects\tauri\target\debug\app-updater.exe" -ArgumentList
+                                                             ~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Start-Process], ParameterBindingException
    + FullyQualifiedErrorId : MissingArgument,Microsoft.PowerShell.Commands.StartProcessCommand
```

can you change the updater to check if the argument list is empty and not add the `-ArgumentList` thing if so?

**Comment by lucasfernog** [comment]
Another crate to downgrade on CI: `memchr` :(

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**